### PR TITLE
connector: try all resolved IP addresses

### LINKED ISSUE #251: try next IP if one fails
today i had an issue where my domain had an ipv6 and an ipv4 response from DNS, but the ipv6 address was blocked by firewall rules and the ipv4 address wasn't tried. If one of the IPs returned from a DNS lookup fails it'd be friendly to alert the caller and/or just try the next ip in the response.

### PR DESCRIPTION
**Pull request by ekoby**
[fixes #251]

### CODE REVIEW

File: src/connector.c [hunk 1]
```diff
@@ -95,50 +113,97 @@ const tlsuv_connector_t* tlsuv_global_connector() {
...
-#endif
 
-    if (r->cancel) {
-        r->error = UV_ECANCELED;
+        if (rc == 0) {
+            UM_LOG(TRACE, "waiting more");
+            continue;
+        }
+
+        for (int i = 0; i < cr->fds; i++) {
```
Comment (dovholuknf) [inline src/connector.c:161]: does it make sense to sort or iterate these randomly before finding an IP to use?

File: src/connector.c [hunk 2]
```diff
@@ -150,36 +215,44 @@ static void on_resolve(uv_getaddrinfo_t *r, int status, struct addrinfo *addr) {
...
-#else
-            e == EINPROGRESS
-#endif
-                ) {
-            uv_poll_init_socket(r->loop, &cr->poll, cr->sock);
-            uv_poll_start(&cr->poll, UV_WRITABLE|UV_DISCONNECT, on_poll_connect);
+    struct addrinfo *addr = addrlist;
+    int count = 0;
+    int err = 0;
+    while (addr && count < max_connect_socks) {
```
Comment (dovholuknf) [inline src/connector.c:227]: if someone adds 17 IPs and the first 16 all fail i feel like at very least we should tell them that all 16 failed and there were 'n' left we didn't use/try. -- i realize this is quite and unlikely situation...

**Review by dovholuknf — APPROVED** [review]
lgtm. couple of questions to answer or ignore... :)

File: src/connector.c (hunk 1 above)
Comment (ekoby) [reply src/connector.c:185]: it won't make a difference -- it tries to connect to all at the same time first to connect is selected

**Pull request merged**